messages.go

```diff
@@ -182,3 +182,126 @@ func NewMsg(msgType string, parent ComposedMsg) (ComposedMsg, error) {
 
 	return msg, nil
 }
+
+// Publish creates a new ComposedMsg and sends it back to the return identities over the
+// IOPub channel.
+func (receipt *msgReceipt) Publish(msgType string, content interface{}) error {
+	msg, err := NewMsg(msgType, receipt.Msg)
+
+	if err != nil {
+		return err
+	}
+
+	msg.Content = content
+	return receipt.SendResponse(receipt.Sockets.IOPubSocket, msg)
+}
+
+// Reply creates a new ComposedMsg and sends it back to the return identities over the
+// Shell channel.
+func (receipt *msgReceipt) Reply(msgType string, content interface{}) error {
+	msg, err := NewMsg(msgType, receipt.Msg)
+
+	if err != nil {
+		return err
+	}
+
+	msg.Content = content
+	return receipt.SendResponse(receipt.Sockets.ShellSocket, msg)
+}
+
+// MIMEDataBundle holds data that can be presented in multiple formats. The keys are MIME types
+// and the values are the data formatted with respect to it's MIME type. All bundle should contain
+// at least a "text/plain" representation with a string value.
+type MIMEDataBundle map[string]interface{}
+
+// NewTextMIMEDataBundle creates a MIMEDataBundle that only contains a text representation described
+// the the parameter 'value'.
+func NewTextMIMEDataBundle(value string) MIMEDataBundle {
+	return MIMEDataBundle{
+		"text/plain": value,
+	}
+}
+
+type KernelStatus string
+
+const (
```

Comment:
I suggest the following minor changes regarding these constants and the `PublishKernelStatus` function:

- Let's let `PublishKernelStatus` accept a string rather than the `KernelStatus` type. Not sure that the type buys us a lot, unless you have something specific in mind.  It doesn't prevent other devs from defining a new KernelStatus that we didn't anticipate, and we are only going to use the constant statuses.
- We can then remove the `KernelStatus` type and move the `const (...)` declarations to the top of `kernel.go`, because they are really relevant to the kernel itself, not the messaging.  
- Lastly, I think we can make the status constants unexported.